### ISSUE TITLE
change how `train` and `learncurve` determine whether dataset has unlabeled segments

### ISSUE THREAD

**Issue by NickleDave** [NickleDave/vak#294]
@yardencsGitHub reports that the start of `learncurve` and `train` is very slow during start-up for him

We noticed it is right after this log statement:
```
        f'Size of each timebin in spectrogram, in seconds: {timebin_dur}',
```

Inspecting code reveals the next step is determining whether the dataset has unlabeled segments.  
Currently this is done using the function `vak.csv.has_unlabeled`
https://github.com/NickleDave/vak/blob/f5520a7acebccea7ca1bb77cd0840b65e32975bd/src/vak/core/learncurve.py#L423
This function does the following: 
- opens every array file containing spectrograms
- makes labeled timebin vectors that are full of a dummy `unlabeled` label, using timebin vector from spectrogram
- uses associated annotation to assign labels to every timebin
- check whether there are still any 'unlabeled' labels in the timebin vector we made, append True or False to a list
- finally return whether there are any True in that list

Obviously this is really expensive because of all the file I/O on giant datasets

One possible fix:
- [ ] just use annotation alone to determine whether there are any unlabeled segments
  - basically just check whether any onset[n+1] - offset[n] > 0 
  - Not obvious to me if I had a good reason to **not** do it this way before. Can't think of why this wouldn't work?
- [ ] if we take this approach, should move function to `annotation` module
- [ ] can still iterate over every annotation, or possibly just return True first time we find unlabeled segments? to be even faster
- [ ] should add logging statements to both `train` and `learncurve` to indicate what's going on
